stdlib/dynArray.ml

```diff
@@ -0,0 +1,130 @@
+let baseLen = 50;;
+let multiplier = 2;;
+
+type 'a cell =
+  | Nil
+  | Cell of 'a
+;;
```

Comment:
This is the option type, and it adds one indirection to every array access. See the vector module in containers to see one way to avoid this issue.